From cea8dbdc0f18367f06756856762feaf340617cea Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sat, 3 Feb 2018 18:42:49 +0000
Subject: [PATCH] allowsharing needs to be set for mounted drives to be
 streamed properly. See
 https://discourse.osmc.tv/t/yatse-and-osmc-2017-12-1-local-streaming/70370/24.
 Pre-define a source and make sure they are always set if there is no
 sources.xml (first-run).

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 userdata/sources.xml             | 21 +++++++++++++++++++++
 xbmc/application/Application.cpp |  1 +
 2 files changed, 22 insertions(+)
 create mode 100644 userdata/sources.xml

diff --git a/userdata/sources.xml b/userdata/sources.xml
new file mode 100644
index 0000000000..1fc93b51e9
--- /dev/null
+++ b/userdata/sources.xml
@@ -0,0 +1,21 @@
+<sources>
+    <video>
+        <default pathversion="1"></default>
+        <source>
+            <name>Auto-mounted drives</name>
+            <path pathversion="1">/media/</path>
+            <allowsharing>true</allowsharing>
+        </source>
+    </video>
+    <music>
+        <default pathversion="1"></default>
+        <source>
+            <name>Auto-mounted drives</name>
+            <path pathversion="1">/media/</path>
+            <allowsharing>true</allowsharing>
+        </source>
+    </music>
+    <files>
+        <default pathversion="1"></default>
+    </files>
+</sources>
diff --git a/xbmc/application/Application.cpp b/xbmc/application/Application.cpp
index 239bf65668..f0489db9ea 100644
--- a/xbmc/application/Application.cpp
+++ b/xbmc/application/Application.cpp
@@ -384,6 +384,7 @@ bool CApplication::Create()
   CUtil::CopyUserDataIfNeeded("special://masterprofile/", "RssFeeds.xml");
   CUtil::CopyUserDataIfNeeded("special://masterprofile/", "favourites.xml");
   CUtil::CopyUserDataIfNeeded("special://masterprofile/", "Lircmap.xml");
+  CUtil::CopyUserDataIfNeeded("special://masterprofile/", "sources.xml");
 
   CServiceBroker::GetLogging().Initialize(CSpecialProtocol::TranslatePath("special://logpath"));
 
-- 
2.25.1

